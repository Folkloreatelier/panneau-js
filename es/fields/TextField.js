import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
import _extends from 'babel-runtime/helpers/extends';
import _defineProperty from 'babel-runtime/helpers/defineProperty';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _createClass from 'babel-runtime/helpers/createClass';
import _inherits from 'babel-runtime/helpers/inherits';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import isNumber from 'lodash/isNumber';
import isString from 'lodash/isString';
import isArray from 'lodash/isArray';
import FormGroup from '../FormGroup';

/**
 *  Class: TextField
 *
 *  @param {string,number,array} value
 *  @return {string} newValue
 */
var propTypes = {
    type: PropTypes.oneOf(['text', 'email', 'password', 'url', 'number', 'textarea', 'editor']),
    name: PropTypes.string,
    label: PropTypes.string,
    placeholder: PropTypes.string,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    onChange: PropTypes.func,
    onFocus: PropTypes.func,
    onBlur: PropTypes.func,

    prefix: PropTypes.node,
    suffix: PropTypes.node,
    prefixClassName: PropTypes.string,
    suffixClassName: PropTypes.string,
    align: PropTypes.oneOf(['left', 'right', 'center']),
    inputOnly: PropTypes.bool,
    disabled: PropTypes.bool,
    ckeditorConfig: PropTypes.object, // eslint-disable-line
    ckeditorCustomConfig: PropTypes.string,
    ckeditorBasePath: PropTypes.string
};

var defaultProps = {
    type: 'text',
    name: null,
    label: null,
    placeholder: null,
    value: null,
    onChange: null,
    onFocus: null,
    onBlur: null,

    prefix: null,
    suffix: null,
    prefixClassName: null,
    suffixClassName: null,
    align: null,
    inputOnly: false,
    disabled: false,
    ckeditorConfig: null,
    ckeditorCustomConfig: null,
    ckeditorBasePath: 'https://cdn.ckeditor.com/4.7.2/standard/'
};

var TextField = function (_Component) {
    _inherits(TextField, _Component);

    _createClass(TextField, null, [{
        key: 'parse',
        value: function parse(value) {
            if (isString(value) || isNumber(value)) {
                return value;
            } else if (isArray(value)) {
                return value.join(' ');
            }
            return ''; // Empty string for the input field
        }
    }]);

    function TextField(props) {
        _classCallCheck(this, TextField);

        var _this = _possibleConstructorReturn(this, (TextField.__proto__ || Object.getPrototypeOf(TextField)).call(this, props));

        _this.onChange = _this.onChange.bind(_this);
        _this.onEditorReady = _this.onEditorReady.bind(_this);
        _this.onEditorChange = _this.onEditorChange.bind(_this);

        _this.editor = null;
        _this.input = null;
        _this.ckeditor = null;
        return _this;
    }

    _createClass(TextField, [{
        key: 'componentDidMount',
        value: function componentDidMount() {
            var _this2 = this;

            var _props = this.props,
                type = _props.type,
                ckeditorBasePath = _props.ckeditorBasePath;

            if (type === 'editor') {
                window.CKEDITOR_BASEPATH = ckeditorBasePath;
                import( /* webpackChunkName: "vendor/ckeditor" */'ckeditor').then(function () {
                    var _props2 = _this2.props,
                        ckeditorConfig = _props2.ckeditorConfig,
                        ckeditorCustomConfig = _props2.ckeditorCustomConfig;

                    _this2.ckeditor = CKEDITOR; // eslint-disable-line no-undef
                    var editor = _this2.ckeditor.replace(_this2.editor, Object.assign({
                        customConfig: ckeditorCustomConfig
                    }, ckeditorConfig || {}));
                    editor.on('instanceReady', _this2.onEditorReady);
                    editor.on('change', _this2.onEditorChange);
                });
            }
        }
    }, {
        key: 'componentWillUnmount',
        value: function componentWillUnmount() {
            var type = this.props.type;

            if (type === 'editor') {
                this.ckeditor.remove(this.editor);
            }
        }
    }, {
        key: 'onEditorReady',
        value: function onEditorReady(e) {
            e.editor.setData(this.props.value);
        }
    }, {
        key: 'onChange',
        value: function onChange(e) {
            var newValue = (e.currentTarget || this.input).value;
            if (this.props.onChange) {
                this.props.onChange(newValue);
            }
        }
    }, {
        key: 'onEditorChange',
        value: function onEditorChange(e) {
            var newValue = e.editor.getData();
            if (this.props.onChange) {
                this.props.onChange(newValue);
            }
        }
    }, {
        key: 'getInput',
        value: function getInput() {
            return this.input;
        }
    }, {
        key: 'renderInput',
        value: function renderInput() {
            var _classNames,
                _this3 = this;

            var _props3 = this.props,
                type = _props3.type,
                name = _props3.name,
                placeholder = _props3.placeholder,
                value = _props3.value,
                align = _props3.align,
                disabled = _props3.disabled,
                onFocus = _props3.onFocus,
                onBlur = _props3.onBlur;

            var inputValue = TextField.parse(value);

            var inputClassNames = classNames((_classNames = {
                'form-control': true
            }, _defineProperty(_classNames, 'field-' + type, true), _defineProperty(_classNames, 'text-' + align, align !== null), _classNames));

            var inputProps = {
                ref: function ref(_ref) {
                    _this3.input = _ref;
                },
                value: inputValue,
                name: name,
                id: name,
                placeholder: placeholder,
                disabled: disabled,
                className: inputClassNames,
                onChange: this.onChange,
                onFocus: onFocus,
                onBlur: onBlur
            };

            var input = null;
            if (type === 'textarea') {
                input = React.createElement('textarea', inputProps);
            } else if (type === 'editor') {
                input = React.createElement(
                    'div',
                    { className: 'editor' },
                    React.createElement('textarea', {
                        className: 'field-editor',
                        name: 'editor',
                        ref: function ref(el) {
                            _this3.editor = el;
                        }
                    }),
                    React.createElement('input', {
                        type: 'hidden',
                        name: name,
                        value: inputValue
                    })
                );
            } else {
                input = React.createElement('input', _extends({}, inputProps, {
                    type: type
                }));
            }
            return input;
        }
    }, {
        key: 'renderInputGroup',
        value: function renderInputGroup(input) {
            var _props4 = this.props,
                type = _props4.type,
                prefix = _props4.prefix,
                suffix = _props4.suffix,
                prefixClassName = _props4.prefixClassName,
                suffixClassName = _props4.suffixClassName;


            var prefixClassNames = classNames(_defineProperty({
                'input-group-addon': prefixClassName === null && isString(prefix),
                'input-group-addon input-group-addon-small': prefixClassName === null && !isString(prefix)
            }, prefixClassName, prefixClassName !== null));

            var suffixClassNames = classNames(_defineProperty({
                'input-group-addon': suffixClassName === null && isString(suffix),
                'input-group-addon input-group-addon-small': suffixClassName === null && !isString(suffix)
            }, suffixClassName, suffixClassName !== null));

            var renderedPrefix = prefix ? React.createElement(
                'span',
                { className: prefixClassNames },
                prefix
            ) : null;

            var renderedSuffix = suffix ? React.createElement(
                'span',
                { className: suffixClassNames },
                suffix
            ) : null;

            var groupClassNames = classNames(_defineProperty({
                'input-group': true,
                'input-group-text': true
            }, 'input-group-' + type, true));

            return React.createElement(
                'div',
                { className: groupClassNames },
                renderedPrefix,
                input,
                renderedSuffix
            );
        }
    }, {
        key: 'render',
        value: function render() {
            var _props5 = this.props,
                label = _props5.label,
                name = _props5.name,
                prefix = _props5.prefix,
                suffix = _props5.suffix,
                other = _objectWithoutProperties(_props5, ['label', 'name', 'prefix', 'suffix']);

            var input = this.renderInput();
            var inputGroup = prefix || suffix ? this.renderInputGroup(input) : input;
            return React.createElement(
                FormGroup,
                _extends({
                    className: 'form-group-text',
                    name: name,
                    label: label
                }, other),
                inputGroup
            );
        }
    }]);

    return TextField;
}(Component);

TextField.propTypes = propTypes;
TextField.defaultProps = defaultProps;

export default TextField;