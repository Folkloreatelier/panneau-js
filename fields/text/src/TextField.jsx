import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import isNumber from 'lodash/isNumber';
import isString from 'lodash/isString';
import isArray from 'lodash/isArray';
import FormGroup from '@panneau/form-group';

/**
 *  Class: TextField
 *
 *  @param {string,number,array} value
 *  @return {string} newValue
 */
const propTypes = {
    type: PropTypes.oneOf([
        'text', 'email', 'password', 'url', 'number', 'textarea', 'editor',
    ]),
    name: PropTypes.string,
    label: PropTypes.string,
    placeholder: PropTypes.string,
    value: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
    ]),
    onChange: PropTypes.func,
    onFocus: PropTypes.func,
    onBlur: PropTypes.func,

    prefix: PropTypes.node,
    suffix: PropTypes.node,
    prefixClassName: PropTypes.string,
    suffixClassName: PropTypes.string,
    align: PropTypes.oneOf([
        'left', 'right', 'center',
    ]),
    inputOnly: PropTypes.bool,
    disabled: PropTypes.bool,
    ckeditorConfig: PropTypes.object, // eslint-disable-line
    ckeditorCustomConfig: PropTypes.string,
    ckeditorBasePath: PropTypes.string,
};

const defaultProps = {
    type: 'text',
    name: null,
    label: null,
    placeholder: null,
    value: null,
    onChange: null,
    onFocus: null,
    onBlur: null,

    prefix: null,
    suffix: null,
    prefixClassName: null,
    suffixClassName: null,
    align: null,
    inputOnly: false,
    disabled: false,
    ckeditorConfig: null,
    ckeditorCustomConfig: null,
    ckeditorBasePath: 'https://cdn.ckeditor.com/4.7.2/standard/',
};

class TextField extends Component {
    static parse(value) {
        if (isString(value) || isNumber(value)) {
            return value;
        } else if (isArray(value)) {
            return value.join(' ');
        }
        return ''; // Empty string for the input field
    }

    constructor(props) {
        super(props);

        this.onChange = this.onChange.bind(this);
        this.onEditorReady = this.onEditorReady.bind(this);
        this.onEditorChange = this.onEditorChange.bind(this);

        this.importCanceled = false;
        this.ckeditor = null;
        this.editor = null;
        this.refInput = null;
        this.refInputGroup = null;
    }

    componentDidMount() {
        const { type, ckeditorBasePath } = this.props;
        if (type === 'editor') {
            window.CKEDITOR_BASEPATH = ckeditorBasePath;
            import(/* webpackChunkName: "vendor/ckeditor" */ 'ckeditor')
                .then(() => {
                    if (this.importCanceled) {
                        return;
                    }
                    const { ckeditorConfig, ckeditorCustomConfig } = this.props;
                    this.ckeditor = CKEDITOR; // eslint-disable-line no-undef
                    const editor = this.ckeditor.replace(this.editor, {
                        customConfig: ckeditorCustomConfig,
                        ...ckeditorConfig || {},
                    });
                    editor.on('instanceReady', this.onEditorReady);
                    editor.on('change', this.onEditorChange);
                });
        }
    }

    componentWillUnmount() {
        const { type } = this.props;
        this.importCanceled = true;
        if (type === 'editor') {
            this.ckeditor.remove(this.editor);
        }
    }

    onEditorReady(e) {
        e.editor.setData(this.props.value);
    }

    onChange(e) {
        const newValue = (e.currentTarget || this.refInput).value;
        if (this.props.onChange) {
            this.props.onChange(newValue);
        }
    }

    onEditorChange(e) {
        const newValue = e.editor.getData();
        if (this.props.onChange) {
            this.props.onChange(newValue);
        }
    }

    getInputRef() {
        return this.refInput;
    }

    getInputGroupRef() {
        return this.refInputGroup;
    }

    renderInput() {
        const {
            type,
            name,
            placeholder,
            value,
            align,
            disabled,
            onFocus,
            onBlur,
        } = this.props;
        const inputValue = TextField.parse(value);

        const inputClassNames = classNames({
            'form-control': true,
            [`field-${type}`]: true,
            [`text-${align}`]: align !== null,
        });

        const inputProps = {
            ref: (ref) => { this.refInput = ref; },
            value: inputValue,
            name,
            id: name,
            placeholder,
            disabled,
            className: inputClassNames,
            onChange: this.onChange,
            onFocus,
            onBlur,
        };

        let input = null;
        if (type === 'textarea') {
            input = (
                <textarea
                    {...inputProps}
                />
            );
        } else if (type === 'editor') {
            input = (
                <div className="editor">
                    <textarea
                        className="field-editor"
                        name="editor"
                        ref={(el) => { this.editor = el; }}
                    />
                    <input
                        type="hidden"
                        name={name}
                        value={inputValue}
                    />
                </div>
            );
        } else {
            input = (
                <input
                    {...inputProps}
                    type={type}
                />
            );
        }
        return input;
    }

    renderInputGroup(input) {
        const {
            type,
            prefix,
            suffix,
            prefixClassName,
            suffixClassName,
        } = this.props;

        const prefixClassNames = classNames({
            'input-group-addon': prefixClassName === null && isString(prefix),
            'input-group-addon input-group-addon-small': prefixClassName === null && !isString(prefix),
            [prefixClassName]: prefixClassName !== null,
        });

        const suffixClassNames = classNames({
            'input-group-addon': suffixClassName === null && isString(suffix),
            'input-group-addon input-group-addon-small': suffixClassName === null && !isString(suffix),
            [suffixClassName]: suffixClassName !== null,
        });

        const renderedPrefix = prefix ?
            <span className={prefixClassNames}>{prefix}</span> : null;

        const renderedSuffix = suffix ?
            <span className={suffixClassNames}>{suffix}</span> : null;

        const groupClassNames = classNames({
            'input-group': true,
            'input-group-text': true,
            [`input-group-${type}`]: true,
        });

        return (
            <div
                className={groupClassNames}
                ref={(ref) => { this.refInputGroup = ref; }}
            >
                { renderedPrefix }
                { input }
                { renderedSuffix }
            </div>
        );
    }

    render() {
        const {
            label,
            name,
            prefix,
            suffix,
            ...other
        } = this.props;
        const input = this.renderInput();
        const inputGroup = prefix || suffix ? this.renderInputGroup(input) : input;
        return (
            <FormGroup
                className="form-group-text"
                name={name}
                label={label}
                {...other}
            >
                { inputGroup }
            </FormGroup>
        );
    }
}

TextField.propTypes = propTypes;
TextField.defaultProps = defaultProps;

export default TextField;
