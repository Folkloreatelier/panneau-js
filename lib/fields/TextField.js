import _extends from 'babel-runtime/helpers/extends';
import _defineProperty from 'babel-runtime/helpers/defineProperty';
import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _createClass from 'babel-runtime/helpers/createClass';
import _inherits from 'babel-runtime/helpers/inherits';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import isNumber from 'lodash/isNumber';
import isString from 'lodash/isString';
import isArray from 'lodash/isArray';
import FormGroup from '../FormGroup';

/**
 *  Class: TextField
 *
 *  @param {string,number,array} value
 *  @return {string} newValue
 */
var propTypes = {
    type: PropTypes.oneOf(['text', 'email', 'password', 'url', 'number', 'textarea', 'editor']),
    name: PropTypes.string,
    label: PropTypes.string,
    placeholder: PropTypes.string,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    onChange: PropTypes.func,

    prefix: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
    suffix: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
    align: PropTypes.oneOf(['left', 'right', 'center']),
    disabled: PropTypes.bool,
    ckeditorCustomConfig: PropTypes.object
};

var defaultProps = {
    type: 'text',
    name: null,
    label: null,
    placeholder: null,
    value: null,
    onChange: null,

    prefix: null,
    suffix: null,
    align: null,
    disabled: false,
    ckeditorCustomConfig: {}
};

var TextField = function (_Component) {
    _inherits(TextField, _Component);

    _createClass(TextField, null, [{
        key: 'parse',
        value: function parse(value) {
            if (isString(value) || isNumber(value)) {
                return value;
            } else if (isArray(value)) {
                return value.join(' ');
            }
            return ''; // Empty string for the input field
        }
    }]);

    function TextField(props) {
        _classCallCheck(this, TextField);

        var _this = _possibleConstructorReturn(this, (TextField.__proto__ || Object.getPrototypeOf(TextField)).call(this, props));

        _this.onChange = _this.onChange.bind(_this);
        _this.onEditorReady = _this.onEditorReady.bind(_this);
        _this.onEditorChange = _this.onEditorChange.bind(_this);

        _this.editor = null;
        _this.ckeditor = null;
        return _this;
    }

    _createClass(TextField, [{
        key: 'componentDidMount',
        value: function componentDidMount() {
            var _this2 = this;

            var type = this.props.type;

            if (type === 'editor') {
                import( /* webpackChunkName: "vendor/ckeditor" */'ckeditor').then(function () {
                    _this2.ckeditor = CKEDITOR; // eslint-disable-line no-undef
                    var editor = _this2.ckeditor.replace('editor');
                    if (_this2.props.ckeditorCustomConfig) {
                        editor.config = _this2.props.ckeditorCustomConfig;
                    }
                    editor.on('instanceReady', _this2.onEditorReady);
                    editor.on('change', _this2.onEditorChange);
                });
            }
        }
    }, {
        key: 'componentWillUnmount',
        value: function componentWillUnmount() {
            var type = this.props.type;

            if (type === 'editor') {
                this.ckeditor.remove(this.editor);
            }
        }
    }, {
        key: 'onEditorReady',
        value: function onEditorReady(e) {
            e.editor.setData(this.props.value);
        }
    }, {
        key: 'onChange',
        value: function onChange(e) {
            var newValue = e.target.value;
            if (this.props.onChange && isString(newValue) && this.props.value !== newValue) {
                this.props.onChange(newValue);
            }
        }
    }, {
        key: 'onEditorChange',
        value: function onEditorChange(e) {
            var newValue = e.editor.getData();
            if (this.props.onChange && isString(newValue) && this.props.value !== newValue) {
                this.props.onChange(newValue);
            }
        }
    }, {
        key: 'render',
        value: function render() {
            var _this3 = this;

            var _props = this.props,
                type = _props.type,
                name = _props.name,
                label = _props.label,
                placeholder = _props.placeholder,
                value = _props.value,
                prefix = _props.prefix,
                suffix = _props.suffix,
                align = _props.align,
                disabled = _props.disabled,
                other = _objectWithoutProperties(_props, ['type', 'name', 'label', 'placeholder', 'value', 'prefix', 'suffix', 'align', 'disabled']);

            var defaultValue = TextField.parse(value);

            var fieldClassNames = classNames(_defineProperty({
                'field-text': true,
                'form-control': true
            }, 'text-' + align, align !== null));

            var field = null;
            if (type === 'textarea') {
                field = React.createElement('textarea', { className: 'field-textarea form-control', name: name, value: defaultValue, onChange: this.onChange, disabled: disabled });
            } else if (type === 'editor') {
                field = React.createElement(
                    'div',
                    _extends({ className: 'editor' }, other),
                    React.createElement('textarea', { id: 'editor', className: 'field-editor', name: 'editor', ref: function ref(el) {
                            _this3.editor = el;
                        } }),
                    React.createElement('input', { type: 'hidden', name: name, value: defaultValue })
                );
            } else {
                field = React.createElement('input', {
                    id: name,
                    type: type,
                    className: fieldClassNames,
                    name: name,
                    value: defaultValue,
                    placeholder: placeholder,
                    disabled: disabled,
                    onChange: this.onChange
                });
            }

            var inputGroup = null;
            if (prefix || suffix) {
                var fixClassNames = classNames({
                    'input-group-addon': isString(suffix),
                    'input-group-addon input-group-addon-small': !isString(suffix)
                });

                var renderedPrefix = prefix ? React.createElement(
                    'span',
                    { className: fixClassNames },
                    prefix
                ) : null;

                var renderedSuffix = suffix ? React.createElement(
                    'span',
                    { className: fixClassNames },
                    suffix
                ) : null;

                var groupClassNames = classNames(_defineProperty({
                    'input-group': true,
                    'input-group-text': true
                }, 'input-group-' + type, true));

                inputGroup = React.createElement(
                    'div',
                    { className: groupClassNames },
                    renderedPrefix,
                    field,
                    renderedSuffix
                );
            } else {
                inputGroup = field;
            }

            return React.createElement(
                FormGroup,
                _extends({ className: 'form-group-text', name: name, label: label }, other),
                inputGroup
            );
        }
    }]);

    return TextField;
}(Component);

TextField.propTypes = propTypes;
TextField.defaultProps = defaultProps;

export default TextField;