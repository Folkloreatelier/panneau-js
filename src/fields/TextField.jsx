import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import isNumber from 'lodash/isNumber';
import isString from 'lodash/isString';
import isArray from 'lodash/isArray';
import FormGroup from '../FormGroup';

/**
 *  Class: TextField
 *
 *  @param {string,number,array} value
 *  @return {string} newValue
 */
const propTypes = {
    type: PropTypes.oneOf(
        ['text', 'email', 'password', 'url', 'number', 'textarea', 'editor'],
    ),
    name: PropTypes.string,
    label: PropTypes.string,
    placeholder: PropTypes.string,
    value: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
    ]),
    onChange: PropTypes.func,

    prefix: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.object,
    ]),
    suffix: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.object,
    ]),
    align: PropTypes.oneOf(
        ['left', 'right', 'center'],
    ),
    inputOnly: PropTypes.bool,
    disabled: PropTypes.bool,
    ckeditorConfig: PropTypes.object, // eslint-disable-line
    ckeditorCustomConfig: PropTypes.string,
    ckeditorBasePath: PropTypes.string,
};

const defaultProps = {
    type: 'text',
    name: null,
    label: null,
    placeholder: null,
    value: null,
    onChange: null,

    prefix: null,
    suffix: null,
    align: null,
    inputOnly: false,
    disabled: false,
    ckeditorConfig: null,
    ckeditorCustomConfig: null,
    ckeditorBasePath: 'https://cdn.ckeditor.com/4.7.2/standard/',
};

class TextField extends Component {
    static parse(value) {
        if (isString(value) || isNumber(value)) {
            return value;
        } else if (isArray(value)) {
            return value.join(' ');
        }
        return ''; // Empty string for the input field
    }

    constructor(props) {
        super(props);

        this.onChange = this.onChange.bind(this);
        this.onEditorReady = this.onEditorReady.bind(this);
        this.onEditorChange = this.onEditorChange.bind(this);

        this.editor = null;
        this.ckeditor = null;
    }

    componentDidMount() {
        const { type, ckeditorBasePath } = this.props;
        if (type === 'editor') {
            window.CKEDITOR_BASEPATH = ckeditorBasePath;
            import(/* webpackChunkName: "vendor/ckeditor" */ 'ckeditor')
                .then(() => {
                    const { ckeditorConfig, ckeditorCustomConfig } = this.props;
                    this.ckeditor = CKEDITOR; // eslint-disable-line no-undef
                    const editor = this.ckeditor.replace(this.editor, {
                        customConfig: ckeditorCustomConfig,
                        ...ckeditorConfig || {},
                    });
                    editor.on('instanceReady', this.onEditorReady);
                    editor.on('change', this.onEditorChange);
                });
        }
    }

    componentWillUnmount() {
        const { type } = this.props;
        if (type === 'editor') {
            this.ckeditor.remove(this.editor);
        }
    }

    onEditorReady(e) {
        e.editor.setData(this.props.value);
    }

    onChange(e) {
        const newValue = e.target.value;
        if (this.props.onChange && isString(newValue) && this.props.value !== newValue) {
            this.props.onChange(newValue);
        }
    }

    onEditorChange(e) {
        const newValue = e.editor.getData();
        if (this.props.onChange && isString(newValue) && this.props.value !== newValue) {
            this.props.onChange(newValue);
        }
    }

    render() {
        const {
            type,
            name,
            label,
            placeholder,
            value,
            prefix,
            suffix,
            align,
            disabled,
            inputOnly,
            ckeditorCustomConfig,
            ...other
        } = this.props;

        const defaultValue = TextField.parse(value);

        const fieldClassNames = classNames({
            'field-text': true,
            'form-control': true,
            [`text-${align}`]: align !== null,
        });

        let input = null;
        if (type === 'textarea') {
            input = <textarea className="field-textarea form-control" name={name} value={defaultValue} onChange={this.onChange} disabled={disabled} />;
        } else if (type === 'editor') {
            input = (
                <div className="editor" {...other}>
                    <textarea className="field-editor" name="editor" ref={(el) => { this.editor = el; }} />
                    <input type="hidden" name={name} value={defaultValue} />
                </div>
            );
        } else {
            input = (
                <input
                    id={name}
                    type={type}
                    className={fieldClassNames}
                    name={name}
                    value={defaultValue}
                    placeholder={placeholder}
                    disabled={disabled}
                    onChange={this.onChange}
                />
            );
        }

        if (inputOnly) {
            return input;
        }

        let inputGroup = null;
        if (prefix || suffix) {
            const fixClassNames = classNames({
                'input-group-addon': isString(suffix),
                'input-group-addon input-group-addon-small': !isString(suffix),
            });

            const renderedPrefix = prefix ?
                <span className={fixClassNames}>{prefix}</span> : null;

            const renderedSuffix = suffix ?
                <span className={fixClassNames}>{suffix}</span> : null;

            const groupClassNames = classNames({
                'input-group': true,
                'input-group-text': true,
                [`input-group-${type}`]: true,
            });

            inputGroup = (
                <div className={groupClassNames}>
                    { renderedPrefix }
                    { input }
                    { renderedSuffix }
                </div>
            );
        } else {
            inputGroup = input;
        }

        return (
            <FormGroup className="form-group-text" name={name} label={label} {...other} >
                { inputGroup }
            </FormGroup>
        );
    }
}

TextField.propTypes = propTypes;
TextField.defaultProps = defaultProps;

export default TextField;
