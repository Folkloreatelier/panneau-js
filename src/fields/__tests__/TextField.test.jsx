import React from 'react';
import renderer from 'react-test-renderer';
import TextField from '../TextField';

test('TextField render a <div />', () => {
    const component = renderer.create(
        <TextField />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
