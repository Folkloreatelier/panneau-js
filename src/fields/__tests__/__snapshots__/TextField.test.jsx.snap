// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="form-group form-group-text"
>
  <label
    className="control-label"
    htmlFor={null}
  >
    Label
  </label>
  <div
    className="form-group-inner"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <input
      className="field-text form-control"
      disabled={false}
      id={null}
      name={null}
      onChange={[Function]}
      placeholder={null}
      type="text"
      value="value"
    />
  </div>
</div>
`;
